src/HomeWork/hw4/MyArrayList.java

```diff
@@ -0,0 +1,111 @@
+package HomeWork.hw4;
+
+import java.util.Arrays;
+
+public class MyArrayList<T> implements MyList<T> {
+    private T[] array;
+    private int size;
+    private int capacity;
+
+    public MyArrayList() {
+        size = 0;
+        capacity = 10;
+        array = (T[]) new Object[capacity];
+    }
+
+    private void increaseCapacity() {
+        capacity = capacity * 3 / 2;
+        array = Arrays.copyOf(array, capacity);
+    }
+
+    @Override
+    public void add(T value) {
+        if (size == capacity) {
+            increaseCapacity();
+        }
+        array[size] = value;
+        size++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (size >= capacity) {
+            increaseCapacity();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
+    }
+
+    @Override
+    public void addAll(MyList<T> list) {
+        T[] addArray = list.toArray();
+        while (list.size() + this.size() > capacity) {
+            increaseCapacity();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(addArray[i]);
+        }
+    }
+
+    @Override
+    public T get(int index) {
+        isIndexCorrect(index);
+        return array[index];
+    }
+
+    @Override
+    public void set(T value, int index) {
+        isIndexCorrect(index);
+        array[index] = value;
+    }
+
+    @Override
+    public T remove(int index) {
+        isIndexCorrect(index);
+        T removeItem = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index -1);
+        array[size - 1] = null;
```

Comment:
Why do you need   array[size - 1] = null; ?

Reply:
because after System.arraycopy(array, index + 1, array, index, size - index -1); the last 2 elements have the same values, so we "delete" the last element

Reply:
ok, this method  will not allow to remove last element in list